spec/models/intervention_spec.rb

```diff
@@ -213,12 +213,38 @@
     context 'when researcher want to assign the intervention to other resarcher' do
       let(:other_user) { create(:user, :confirmed, :researcher) }
       let(:params) { { emails: [other_user.email.upcase] } }
+      let(:message_delivery) { instance_double(ActionMailer::MessageDelivery) }
+
+      before do
+        allow(message_delivery).to receive(:deliver_now)
+        ActiveJob::Base.queue_adapter = :test
+      end
 
       it 'create a new intervention with correct user_id' do
         cloned_intervention = intervention.clone(params: params)
 
         expect(cloned_intervention.first.user_id).to eq(other_user.id)
       end
+
```

Comment:
Let's add a test if a new user was created (you can use `change(User, :count).by(1)`

Reply:
I don't think the current implementation allows for a simple check of the user count, as there are actually three cases for external sharing:

1. The invited researcher already has a confirmed account => The default email is sent and no new user is created
2. The invited researcher has an account, but it's not confirmed => The email with the invitation to registration is sent, but still, no new user is created (as it already existed, though not confirmed)
3. The invited researcher doesn't have an account => The email with the invitation to registration is sent, **and a new user account is created**

So, I believe I should only check for the change of the user count having checked the conditions and ensured that it's particularly tested only for the 3rd case